File: bestfitpy/tests/test_gev.py
# Binding-specific behaviour (vectorisation, return types). Oracle-value checks
# live in test_fixtures.py, driven by the shared language-neutral fixtures.
import numpy as np

from bestfitpy import dgev, pgev, qgev


def test_vectorised_over_arrays():
    x = np.array([90.0, 100.0, 110.0])
    out = dgev(x, 100, 10, 0)
    assert out.shape == (3,)
    assert out[1] == dgev(100.0, 100, 10, 0)


def test_scalar_returns_scalar():
    assert np.isscalar(pgev(100.0, 100, 10, 0)) or np.ndim(pgev(100.0, 100, 10, 0)) == 0
    assert np.ndim(qgev(0.5, 100, 10, 0)) == 0

File: corehydror/tests/testthat/test-gev.R
# Binding-specific behaviour (vectorisation, return shapes). Oracle-value checks
# live in test-fixtures.R, driven by the shared language-neutral fixtures.

test_that("density, distribution, quantile are vectorised", {
  x <- c(90, 100, 110)
  expect_length(dgev(x, 100, 10, 0), 3L)
  expect_equal(dgev(x, 100, 10, 0)[2], dgev(100, 100, 10, 0))
  expect_length(pgev(x, 100, 10, 0), 3L)
})

test_that("gev_fit returns a named location/scale/shape vector", {
  fit <- gev_fit(c(5, 7, 3, 9, 6, 8, 4, 11, 2, 10), method = "lmom")
  expect_named(fit, c("location", "scale", "shape"))
})
